tests/lib/rules/require-valid-default-prop.js

```diff
@@ -0,0 +1,378 @@
+      code: `export default {
+        props: {
+          foo: {
+            type: Boolean,
+            default: {}
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('boolean')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Boolean,
+            default: []
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('boolean')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: ''
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: 55
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: false
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: {}
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Object,
+            default: []
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: ''
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: 55
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: false
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: {}
+          }
+        }
+      }`,
+      parserOptions,
+      errors: errorMessage('function')
+    },
+    {
+      filename: 'test.vue',
+      code: `export default {
+        props: {
+          foo: {
+            type: Array,
+            default: []
```

Comment:
for now i'm not checking what function is returning or even if function is returning something